converter/dicom-cast/src/Microsoft.Health.DicomCast.Core/Features/Worker/DicomCastWorker.cs

```diff
@@ -73,6 +84,14 @@ public async Task ExecuteAsync(CancellationToken cancellationToken)
                     LogWorkerCancelRequestedDelegate(_logger, null);
                     break;
                 }
+                catch (Exception ex)
+                {
+                    LogUnexpectedExceptionDelegate(_logger, ex);
+
+                    // Any exception in ExecuteAsync will not shutdown application, call hostApplicationLifetime.StopApplication() to force shutdown.
+                    // Please refer to .net core issue on github for more details: "Exceptions in BackgroundService ExecuteAsync are (sometimes) hidden" https://github.com/dotnet/extensions/issues/2363
+                    _hostApplicationLifetime.StopApplication();
```

Comment:
github issue is stopping the finally block and has no code after that.

We have line#97 trying to log more

Reply:
looks like hostApplicationLifetime.StopApplication() send a stop request, what I see in the log is:

```
crit: Microsoft.Health.DicomCast.Core.Features.Worker.DicomCastWorker[0]
      Unhandled exception.
Microsoft.Health.Dicom.Client.DicomWebException: BadGateway
   at Microsoft.Health.Dicom.Client.DicomWebClient.EnsureSuccessStatusCodeAsync(HttpResponseMessage response) in C:\Users\penche\source\repos\dicom-server\src\Microsoft.Health.Dicom.Client\DicomWebClient.cs:line 437
   at Microsoft.Health.Dicom.Client.DicomWebClient.GetChangeFeed(String queryString, CancellationToken cancellationToken) in C:\Users\penche\source\repos\dicom-server\src\Microsoft.Health.Dicom.Client\DicomWebClient.cs:line 267
   at Microsoft.Health.DicomCast.Core.Features.DicomWeb.Service.ChangeFeedRetrieveService.RetrieveChangeFeedAsync(Int64 offset, CancellationToken cancellationToken) in C:\Users\penche\source\repos\dicom-server\converter\dicom-cast\src\Microsoft.Health.DicomCast.Core\Features\DicomWeb\Service\ChangeFeedRetrieveService.cs:line 35
   at Microsoft.Health.DicomCast.Core.Features.Worker.ChangeFeedProcessor.ProcessAsync(TimeSpan pollIntervalDuringCatchup, CancellationToken cancellationToken) in C:\Users\penche\source\repos\dicom-server\converter\dicom-cast\src\Microsoft.Health.DicomCast.Core\Features\Worker\ChangeFeedProcessor.cs:line 56
   at Microsoft.Health.DicomCast.Core.Features.Worker.LoggingChangeFeedProcessor.ProcessAsync(TimeSpan pollIntervalDuringCatchup, CancellationToken cancellationToken) in C:\Users\penche\source\repos\dicom-server\converter\dicom-cast\src\Microsoft.Health.DicomCast.Core\Features\Worker\LoggingChangeFeedProcessor.cs:line 41
   at Microsoft.Health.DicomCast.Core.Features.Worker.DicomCastWorker.ExecuteAsync(CancellationToken cancellationToken) in C:\Users\penche\source\repos\dicom-server\converter\dicom-cast\src\Microsoft.Health.DicomCast.Core\Features\Worker\DicomCastWorker.cs:line 77
info: Microsoft.Hosting.Lifetime[0]
      Application is shutting down...
info: Microsoft.Health.DicomCast.Core.Features.Worker.DicomCastWorker[0]
      Microsoft.Health.DicomCast.Core.Features.Worker.DicomCastWorker is requested to be stopped.
info: Microsoft.Health.DicomCast.Core.Features.Worker.DicomCastWorker[0]
      Microsoft.Health.DicomCast.Core.Features.Worker.DicomCastWorker is exiting.
```

Reply:
StopApplication is good. I am wondering about the code after it. Looks like we should not have any code once we call that, right now we have LogWorkerExitingDelegate

Reply:
The method name is a little bit confusing -- it actually cancels the CancellationTokenSource instead of shutdown host forcedly. In our case, it will go back to while loop, and goes to line "catch (OperationCanceledException) when (cancellationToken.IsCancellationRequested)", then jump out of while loop, and log worker existing.

Reply:
link: https://github.com/dotnet/runtime/blob/3d9bef939ad4edb0aff6cca1c0bcb0e6e7eb3d39/src/libraries/Microsoft.Extensions.Hosting/src/Internal/ApplicationLifetime.cs#L49